pkg/controller/job/job_controller.go

```diff
@@ -566,10 +571,32 @@ func (jm *Controller) syncJob(key string) (bool, error) {
 			}
 		}
 		if complete {
-			job.Status.Conditions = append(job.Status.Conditions, newCondition(batch.JobComplete, "", ""))
+			job.Status.Conditions = append(job.Status.Conditions, newCondition(batch.JobComplete, v1.ConditionTrue, "", ""))
+			jobConditionsChanged = true
 			now := metav1.Now()
 			job.Status.CompletionTime = &now
 			jm.recorder.Event(&job, v1.EventTypeNormal, "Completed", "Job completed")
+		} else if utilfeature.DefaultFeatureGate.Enabled(features.SuspendJob) && manageJobCalled {
+			// Update the conditions / emit events only if manageJob was called in
+			// this syncJob. Otherwise wait for the right syncJob call to make
+			// updates.
+			if job.Spec.Suspend {
+				// Job can be in the suspended state only if it is NOT completed.
+				var isUpdated bool
+				job.Status.Conditions, isUpdated = ensureJobConditionStatus(job.Status.Conditions, batch.JobSuspended, v1.ConditionTrue)
```

Comment:
```suggestion
				job.Status.Conditions, jobConditionsChanged = ensureJobConditionStatus(job.Status.Conditions, batch.JobSuspended, v1.ConditionTrue)
```

Reply:
and remove `isUpdated` and `jobConditionsChanged = true` below

Reply:
what if `jobConditionsChanged` was true before calling `ensureJobConditionStatus`?

Reply:
is that possible?

Reply:
you're right, it's not

Reply:
yeah, ideally we should be trying to set the condition in one place in a sync cycle.

Reply:
Consider a situation where a job failed in this syncJob cycle (jobConditionsChanged set to true in the Failed branch) but the job 
has never been suspended. Then ensureJobConditionStatus's second return val will be false because the job has never been suspended (this is normal), but this would overwrite the JobFailed condition changed status so the controller won't write the update.

Or do you mean changing only the suspend=true branch (which would be the worst of both worlds IMO)? I genuinely think explicitly receiving isUpdated and updating jobConditionsChanged separately is cleaner.

Reply:
> Or do you mean changing only the suspend=true branch (which would be the worst of both worlds IMO)? I genuinely think explicitly receiving isUpdated and updating jobConditionsChanged separately is cleaner.

I mean the exact change I am proposing, nothing more. It is a nit, so up to you, but I find it redundant the way it is done.

Reply:
> > Consider a situation where a job failed in this syncJob cycle (jobConditionsChanged set to true in the Failed branch) but the job has never been suspended. Then ensureJobConditionStatus's second return val will be false because the job has never been suspended (this is normal), but this would overwrite the JobFailed condition changed status so the controller won't write the update.

If you get into that branch, you will never reach here.